### PR TITLE
Carrier signup

### PR DESCRIPTION
**Pull request by dwilkie**

### CODE REVIEW

File: app/controllers/concerns/user_authorization.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ module UserAuthorization
   private
 
   def authorize_user!
-    authorize(@record, policy_class: policy_class)
+    authorize(@record, policy_class:)
```
Comment (samnang) [inline app/controllers/concerns/user_authorization.rb:21]: fancy 👍

File: app/forms/carrier_form.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  validates :name,
+            :work_email,
+            :website,
+            :company,
+            :country,
+            :password,
+            :password_confirmation,
+            presence: true
+
+  validates :work_email, email_format: true, email_uniqueness: true, allow_nil: true, allow_blank: true
```
Comment (samnang) [inline app/forms/carrier_form.rb:25]: I thought `allow_blank` should cover `allow_nil`?

File: app/forms/carrier_form.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  URL_FORMAT = /\A#{URI::DEFAULT_PARSER.make_regexp(%w[https http])}\z/
+
+  attribute :company
+  attribute :name
+  attribute :work_email
+  attribute :country
+  attribute :website
+  attribute :password
+  attribute :password_confirmation
+  attribute :user, default: -> { User.new }
```
Comment (samnang) [inline app/forms/carrier_form.rb:14]: Is it neccessary to have the default value?

**Review by samnang — COMMENTED** [review]
Our dashboard keeps growing in terms of functionalities. What do you think If we split our routes into two namespaces (`carrier` and `account`)?

**Comment by dwilkie** [comment]
Yeah could be. Although we might want to be sure that carrier is the right term. It refers to carriers, BYOC customers (customers who bring their own carrier) and later network providers.

File: app/controllers/concerns/user_authorization.rb (hunk 1 above)
Comment (dwilkie) [reply app/controllers/concerns/user_authorization.rb:21]: អា​នគរបាល​

**Pull request merged**